Mathlib/Data/MvPolynomial/Division.lean

```diff
@@ -264,4 +344,32 @@ theorem X_dvd_monomial {i : σ} {j : σ →₀ ℕ} {r : R} :
 set_option linter.uppercaseLean3 false in
 #align mv_polynomial.X_dvd_monomial MvPolynomial.X_dvd_monomial
 
+lemma X_add_C_ne_zero (i : σ) (c : R) [Nontrivial R] : X i + C c ≠ 0 := by
+  classical
+  intro r
+  have := (ext_iff _ _).mp r (Finsupp.single i 1)
+  rw [coeff_add, coeff_X, coeff_zero, coeff_C, if_neg, add_zero] at this
+  · exact one_ne_zero' R  this
+  · intro rid
+    simpa using FunLike.ext_iff.mp rid i
+
+lemma X_sub_C_ne_zero (i : σ) (c : R') [Nontrivial R'] : X i - C c ≠ 0 := by
+  convert X_add_C_ne_zero i (- c) using 1
+  rw [C_neg, sub_eq_add_neg]
```

Comment:
These don't belong in this file, as far as I can tell they don't use division at all.

Reply:
moved to other files